geomstats/geometry/fiber_bundle.py

```diff
@@ -381,3 +381,48 @@ def integrability_tensor(self, tangent_vec_a, tangent_vec_b, base_point):
         https://doi.org/10.1307/mmj/1028999604.
         """
         raise NotImplementedError
+
+    def integrability_tensor_derivative(self, hor_tg_vec_x, hor_tg_vec_y,
+                                        nabla_x_y, tg_vec_e, nabla_x_e,
+                                        base_point):
+        r"""Compute the covariant derivative of the integrability tensor A.
+
+        The covariant derivative :math: `\nabla_X^M (A_Y E)` in top space is
```

Comment:
Right. The connection is clearly the one of the total space here.
For the shape of arguments, should we keep shape=[..., {ambient_dim, [n, n]}] or use shape=[..., dim] (this applies to several methods)?
ambient_dim is not defined in fiber_bundles (the dimension is dim) and whenever the total space is a matrix space (like for instance in  kendall shape spaces), matrices may not be square.

Reply:
The dimension `dim` in the fiber bundle's `__init__` refers to the dimension of the base space, so we should maybe avoid it?

Maybe: `shape=[..., {total_space.dim, [n, m]}]`?

Reply:
Actually, what we need to refer to would rather be something like `object_shape` (or perhaps here `total_space.shape`), without caring at this level if the total_space is encoded as a vector or a matrix. Using plain English words would also be more understandable than enumerating all possibilities in an algorithmic way, especially since this enumeration will inevitably be incomplete when we will add a new class encoded with a new type of shape (e.g. n-th order tensor). 

Wouldn't it be interesting to add a field in object classes that encodes the shape that this object is supposed to have? For zero tangent vectors, for instance, I would prefer to use `gs.zeros_like(self.shape)` rather than referring to the shape of one of the arguments as I had to do in the `curvature_derive ` method of the Hypershpere class for instance.

Reply:
I have changed the shape specification into shape=[..., {total_space.dim, [n, m]}] as suggested. However, I believe that this is not ]clear for an uninformed user as n and m are not defined in fiber_bundle (and they should not be as the object is abstract at this level).